### PR TITLE
Implement reportUncheckedExceptionDeadCatch

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan/issues/9921

### CODE REVIEW

File: src/Rules/Exceptions/CatchWithUnthrownExceptionRule.php [hunk 1]
```diff
@@ -32,6 +40,14 @@ public function processNode(Node $node, Scope $scope): array
 			];
 		}
 
+		$isCheckedException = TrinaryLogic::createFromBoolean($this->reportUncheckedExceptionDeadCatch)->lazyOr(
```
Comment (ondrejmirtes) [inline src/Rules/Exceptions/CatchWithUnthrownExceptionRule.php]: This code isn't very readable to me. We want to early-return only if `$this->reportUncheckedExceptionDeadCatch` is false. So please do all this logic inside `if (!$this->reportUncheckedExceptionDeadCatch) { ... } `.

Thank you.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Much better, thank you. Please also send an update to the documentation and the blogpost.

**Comment by ondrejmirtes** [comment]
Still waiting for that docs update :)

**Comment by VincentLanglet** [comment]
> Still waiting for that docs update :)

Sorry for the delay, it's definetly in my TODO

**Comment by VincentLanglet** [comment]
I did https://github.com/phpstan/phpstan/pull/9944 @ondrejmirtes

**Pull request merged**